src/message/message.cc

```diff
@@ -28,19 +28,21 @@ void CRMessageBase::Dispatch(const CRMessageBasePtr& message) {
         queue->AddMessage(CRMessageBasePtr(message));
         node->Kick();
     }
+    subscription_info->latest_delivered_time_.store(GetSystemTimestampNsec());
 }
 
 void CRMessageBase::Subscribe(const std::string& message_type, CRNodeBase* node) {
-    impl::subscription_map[message_type].emplace_back(node);
+    (*GetSubscriptionMap())[message_type].sub_list_.emplace_back(node);
 }
 
 void CRMessageBase::Unsubscribe(const std::string& message_type, CRNodeBase* node) {
-    auto subscription_map_search = impl::subscription_map.find(message_type);
-    if (subscription_map_search == impl::subscription_map.end()) {
+    auto* subscription_map        = GetSubscriptionMap();
+    auto  subscription_map_search = subscription_map->find(message_type);
```

Comment:
const?

Reply:
We need to remove items from the list subscription list, so they cannot be const